src/elastio-snap.c

```diff
@@ -4290,22 +4313,30 @@ static int __tracer_setup_tracing(struct snap_device *dev, unsigned int minor){
 	ret = __tracer_transition_tracing(dev, dev->sd_base_dev, tracing_mrf, &snap_devices[minor]);
 #else
 	if (!dev->sd_tracing_ops) {
-		ret = find_orig_fops(dev->sd_base_dev, &dev->sd_orig_ops, &dev->sd_orig_mrf);
+		// Multiple devices on the same disk are sharing block_device_operations struct.
+		// The next call will set a pointer to the dev->sd_tracing_ops if some device at the same disk is
+		// already tracked by the driver. And we'll reuse the existing struct in this case.
+		ret = find_orig_fops(dev->sd_base_dev, &dev->sd_orig_ops, &dev->sd_orig_mrf, &dev->sd_tracing_ops);
 		if(ret) goto error;
 
-		LOG_DEBUG("allocating tracing ops for device with minor %i", minor);
-		dev->sd_tracing_ops = kmalloc(sizeof(struct block_device_operations), GFP_KERNEL);
 		if (!dev->sd_tracing_ops) {
-			ret = -ENOMEM;
-			LOG_ERROR(ret, "error allocating tracing ops");
-			goto error;
-		}
+			LOG_DEBUG("allocating tracing ops for device with minor %i", minor);
+			dev->sd_tracing_ops = kmalloc(sizeof(struct block_device_operations), GFP_KERNEL);
+			if (!dev->sd_tracing_ops) {
+				ret = -ENOMEM;
+				LOG_ERROR(ret, "error allocating tracing ops");
+				goto error;
+			}
 
```

Comment:
Why does the indentation look off here?  Is it an artifact of the GitHub UI or did you really indent this code more than the surrounding code?

Reply:
Probably it's an artifact of the GitHub UI.  The tabs are used in this source file for indentation.  I like more to use spaces, because tabs may look different in different editors and UIs.  But for the code stile consistency we are still using tabs here.
The max indent is 5 tabs on lines 4326-4328.  And it's hard to tell how many spaces are used for 1 tab here in the GitHub Sometimes it looks like 6 and sometimes like 7 spaces. :man_shrugging: